pkg/cmd/grafana-cli/commands/install_command.go

```diff
@@ -111,14 +111,19 @@ func InstallPlugin(pluginName, version string, c utils.CommandLine) error {
 	// Create temp file for downloading zip file
 	tmpFile, err := ioutil.TempFile("", "*.zip")
 	if err != nil {
-		errutil.Wrap("Failed to create temporary file", err)
+		return errutil.Wrap("Failed to create temporary file", err)
 	}
 	defer os.Remove(tmpFile.Name())
+	defer tmpFile.Close()
 
 	err = c.ApiClient().DownloadFile(pluginName, tmpFile, downloadURL, checksum)
 	if err != nil {
 		return errutil.Wrap("Failed to download plugin archive", err)
 	}
+	err = tmpFile.Close()
```

Comment:
I think it's cleaner if we use the technique of a deferred closure around `tmpFile.Close` that sets the named error return value, if nil and `tmpFile.Close` returns an error.

Reply:
Actually on re-inspecting the whole logic, I think we might want to close the file before calling `extractFiles`,  to ensure it can be safely read within`extractFiles`? From what I recall, a file can only be opened once at a time on Windows, so we shouldn't keep it open for write while trying to read from it. What do you think?

If we go with this logic, I think we can also safely drop the deferred closing of the file, since the logic for closing the file is so obvious and close to the site of creating the file.